diff --git a/cmake/templates/_setup_util.py.in b/cmake/templates/_setup_util.py.in
index b08a57a2..3fed2865 100755
--- a/cmake/templates/_setup_util.py.in
+++ b/cmake/templates/_setup_util.py.in
@@ -52,12 +52,13 @@ IS_WINDOWS = (system == 'Windows')
 # subfolder of workspace prepended to CMAKE_PREFIX_PATH
 ENV_VAR_SUBFOLDERS = {
     'CMAKE_PREFIX_PATH': '',
-    'LD_LIBRARY_PATH' if not IS_DARWIN else 'DYLD_LIBRARY_PATH': @CATKIN_LIB_ENVIRONMENT_PATHS@,
     'PATH': '@CATKIN_GLOBAL_BIN_DESTINATION@',
     'PKG_CONFIG_PATH': @CATKIN_PKGCONFIG_ENVIRONMENT_PATHS@,
     'PYTHONPATH': '@PYTHON_INSTALL_DIR@',
 }
 
+if not IS_DARWIN:
+    ENV_VAR_SUBFOLDERS['LD_LIBRARY_PATH'] = @CATKIN_LIB_ENVIRONMENT_PATHS@
 
 def rollback_env_variables(environ, env_var_subfolders):
     '''
@@ -269,14 +270,14 @@ if __name__ == '__main__':
             # don't consider any other prefix path than this one
             CMAKE_PREFIX_PATH = []
         # prepend current workspace if not already part of CPP
-        base_path = os.path.dirname(__file__)
+        # base_path = os.path.dirname(__file__)
         # CMAKE_PREFIX_PATH uses forward slash on all platforms, but __file__ is platform dependent
         # base_path on Windows contains backward slashes, need to be converted to forward slashes before comparison
-        if os.path.sep != '/':
-            base_path = base_path.replace(os.path.sep, '/')
+        # if os.path.sep != '/':
+        #    base_path = base_path.replace(os.path.sep, '/')
 
-        if base_path not in CMAKE_PREFIX_PATH:
-            CMAKE_PREFIX_PATH.insert(0, base_path)
+        # if base_path not in CMAKE_PREFIX_PATH:
+        #     CMAKE_PREFIX_PATH.insert(0, base_path)
         CMAKE_PREFIX_PATH = os.pathsep.join(CMAKE_PREFIX_PATH)
 
         environ = dict(os.environ)
